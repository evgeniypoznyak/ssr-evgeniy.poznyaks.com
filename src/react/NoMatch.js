import React from 'react';

/**
 *
 * @return {*}
 * @constructor
 */
export default function NoMatch() {
    return (
        <div>
            Four Oh Four
        </div>
    );
}
