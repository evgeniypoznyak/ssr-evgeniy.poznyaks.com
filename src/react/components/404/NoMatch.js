import React from 'react';
import {Typography} from '@material-ui/core';

export default function NoMatch(props) {
    return (
        <div>
            <Typography variant="h1" component="h2">
                404: Page not found
            </Typography>
        </div>
    );
}
