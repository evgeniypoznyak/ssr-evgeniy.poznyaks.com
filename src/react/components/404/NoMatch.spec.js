import React from 'react';
import {shallow} from 'enzyme';
import NoMatch from './NoMatch';

let wrapper;

beforeAll(() =>{
    wrapper = shallow(<NoMatch/>);
});

describe('<NoMatch/>', () => {
    it('should render correctly with enzyme', () => {
        expect(wrapper.children.length).toBe(1);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
    it('should be text with "404" and child == 1 in 404 page', () => {
        expect(wrapper.text()).toMatch('404');
    });
});
