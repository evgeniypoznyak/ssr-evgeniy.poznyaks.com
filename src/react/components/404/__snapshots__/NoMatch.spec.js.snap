// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoMatch/> should render correctly with enzyme 1`] = `
<div>
  <ForwardRef(WithStyles)
    component="h2"
    variant="h1"
  >
    404: Page not found
  </ForwardRef(WithStyles)>
</div>
`;
